dags/database/db_context.py

```diff
@@ -0,0 +1,175 @@
+from typing import Any
+import psycopg2
+from psycopg2.extensions import ISOLATION_LEVEL_AUTOCOMMIT
+import os
+import random
+import string
+from returns.pipeline import pipe
```

Comment:
I decided to use returns and ramda to achieve a point free functional style where possible. However, since typing lambda functions (and lambda functions in general) are somewhat limited and clumsy in Python, I still decided to stick with `def` to define functions in most cases.